src/core/classification/qgsclassificationmethod.cpp

```diff
@@ -161,6 +161,42 @@ QString QgsClassificationMethod::formatNumber( double value ) const
   }
 }
 
+void QgsClassificationMethod::addParameter( QgsProcessingParameterDefinition *definition )
+{
+  mParameters.append( definition );
+}
+
+QVariant QgsClassificationMethod::parameterValue( const QString &name ) const
+{
+  const QgsProcessingParameterDefinition *def = parameterDefinition( name );
+  if ( def )
+    return mParameterValues.value( name, def->defaultValue() );
+  else
+    return mParameterValues.value( name );
+}
+
+const QgsProcessingParameterDefinition *QgsClassificationMethod::parameterDefinition( const QString &parameterName ) const
+{
+  for ( const QgsProcessingParameterDefinition *def : mParameters )
+  {
+    if ( def->name() == parameterName )
+      return def;
+  }
+  return nullptr;
+}
+
+void QgsClassificationMethod::setParameterValues( const QVariantMap &values )
+{
+  mParameterValues = values;
+  for ( const QString &paramName : mParameterValues.keys() ) // todo is this ok?
```

Comment:
no, better to iterate over the map in place and avoid the double-iteration required if you call .keys()
```suggestion
  for ( auto it =mParameterValues.begin(); it != mParameterValue.end(); ++it  )
```